### DISCUSSION TITLE
The property "includes" does not exist for the type "never"

### DISCUSSION THREAD

**Discussion by maxfrischknecht** [nuxt/nuxt#18962] [Questions] [answered]
I'm pretty new to typescript and I didn't find any solution for this, maybe it's just a small thing.

I'm trying to test for items from a API with `v-if`:

```ts
<script setup lang="ts">
    const { data: items, pending, refresh, error } = await useFetch(() => 'https://example.ch/api/items?fulltext_search=test');
</script>

<template>
      <div v-for="item in items">
        <template v-if="item['@type'].includes('dctype:Image')">
          <img class="w-full" :href="item['schema:image'][0]['@id'] + '/full/480,/0/default.jpg'">
        </template>
      </div>
</template>
```

The variable `item['@type']` clearly is an array and it contains `[ "o:Item", "dctype:Image" ]`.
Why does typescript tell me `The property "includes" does not exist for the type "never"`?

Thank you for any help, very appreciated!

**Comment by TheAlexLichter** [comment, marked as answer]
You want to type the data from `useFetch` when using external sources (read: "not nuxt server") and then pass it as generic to `useFetch`. Example:

```ts
type ImageResponse {
  '@type': string
  // add more here
}

const { data: items, pending, refresh, error } = await useFetch<ImageResponse>(() => 'https://example.ch/api/items?fulltext_search=test');
```

**Reply by maxfrischknecht** [reply]
Thx for the tipp! I ended up using 

```ts
type ImageResponse = {
  '@type': any
}
```

Because `string` led to new errors.
Thx!

**Reply by TheAlexLichter** [reply]
I'd suggest to make the types as strict as possible (ideally not any 😛) to catch issues on build time 😋

**Reply by maxfrischknecht** [reply]
Makes sense, but how would I do that with a JSON response from a API that contains Strings, Objects, Arrays and Numbers? 🙂

**Reply by TheAlexLichter** [reply]
Well you can define nested types as well. Brief examples from [Kent's blogpost](https://kentcdodds.com/blog/using-fetch-with-type-script#using-the-typed-fetchpokemon):

```ts
type PokemonData = {
  id: string
  number: string
  name: string
  image: string
  fetchedAt: string
  attacks: {
    special: Array<{
      name: string
      type: string
      damage: number
    }>
  }
}
```

**Reply by maxfrischknecht** [reply]
Awesome, thank you a lot! 🙏
